base/bitarray.jl

```diff
@@ -251,6 +251,15 @@ function pack8bools(z::UInt64)
     return z
 end
 
+@inline function unsafe_unaligned_u64_getindex(p::Ptr{Bool}, idx::Int)
+    p64 = Ptr{UInt64}(p)
+    llvmcall("""
+    %p = getelementptr i64, i64* %0, i64 %1
+    %v = load i64, i64* %p, align 1
+    ret i64 %v
+    """, UInt64, Tuple{Ptr{UInt64},Int64}, p64, Int64(idx - 1))
```

Comment:
Ah right – I forgot that I originally didn't trust users / C libraries to declare types exactly correct :P (and then the align attribute was added because the GPU folks complained that it was being too pessimistic)